psyneulink/core/compositions/composition.py

```diff
@@ -9037,16 +9204,19 @@ def run(
                         )
                     node.initialize(initialize_cycle_values[node], context)
 
-        if reset_stateful_functions_to is None:
-            reset_stateful_functions_to = {}
-
-        for node in reset_stateful_functions_to:
-            # make sure the args to reset are in the form of an iterable so they can be unpacked
-            try:
-                iter(reset_stateful_functions_to[node])
-            except TypeError:
-                reset_stateful_functions_to[node] = [reset_stateful_functions_to[node]]
-            node.reset(*reset_stateful_functions_to[node], context=context)
+        try:
+            for node in reset_stateful_functions_to:
```

Comment:
```
for node, vals in reset_state_functions_to.items():
     ...
```
can avoid several repeated lookups

Reply:
updated in ffad3efb0747fedd8306bad83a2bfc5dd5ddee29